gdxsv/lbs_discord.go

```diff
@@ -0,0 +1,494 @@
+	RenpoRoomPeers string
+	ZeonRoomPeers  string
+	NoForcePeers   string
+}
+
+type battlePeers struct {
+	RegionName string
+	RenpoPeers string
+	ZeonPeers  string
+}
+
+func contains(s []string, e string) bool {
+	for _, a := range s {
+		if a == e {
+			return true
+		}
+	}
+	return false
+}
+
+func sortedKeys(m map[uint16]*lobbyPeers) []uint16 {
+	keys := make([]uint16, len(m))
+	i := 0
+	for k := range m {
+		keys[i] = k
+		i++
+	}
+	sort.Slice(keys, func(i, j int) bool { return keys[i] < keys[j] })
+	return keys
+}
+
+//insert Braille Pattern Blank to create newline at ending
+func addBlankIfRequired(s string) string {
+	if len(s) > 0 {
+		return s + "⠀\n"
+	}
+	return s
+}
+
+//Embed limits: Field's value is limited toto 1024 characters
+func splitEmbedFieldValues(value string) []string {
+	var values []string
+
+	//Split field values into 1024 chunks
+	numberOfSplit := int(math.Ceil(float64(len(value)) / 1024))
+
+	splitIndex := 0
+	for i := 0; i < numberOfSplit; i++ {
+		splitIndex = len(value) / (numberOfSplit - i)
+		firstHalf := value[:splitIndex]
+		splitIndex = strings.LastIndex(firstHalf, "\n")
+
+		values = append(values, value[:splitIndex])
+		// splitIndex += 2
+		value = value[splitIndex:]
+	}
+	//do not create empty fields
+	if value != "\n" {
+		values = append(values, value)
+	}
+
+	return values
+}
+
+//Embed limits: There can be up to 25 fields
+func splitEmbedFields(fields []*discordEmbedField) [][]*discordEmbedField {
+	var divided [][]*discordEmbedField
+	chunkSize := 25
+
+	for i := 0; i < len(fields); i += chunkSize {
+		end := i + chunkSize
+
+		if end > len(fields) {
+			end = len(fields)
+		}
+
+		divided = append(divided, fields[i:end])
+	}
+	return divided
+}
+
+func reducePeerStringSize(s string) string {
+	//Reduce size by removing userid
+	re := regexp.MustCompile("`.*?`\\s")
+	s = re.ReplaceAllString(s, "")
+
+	//Reduce size by replacing custom emoji (from 28 char to 1 char)
+	s = strings.ReplaceAll(s, "<:gundam:772467554160738355>", "🌎")
+	s = strings.ReplaceAll(s, "<:zaku:772467605008023563>", "🪐")
+
+	return s
+}
+
+//PublishStatusToDiscord : Update server status to the predefined Discord message thru Web Hook
+func (lbs *Lbs) PublishStatusToDiscord() {
+	if len(conf.DiscordWebhookURL) == 0 {
+		return
+	}
+
+	if atomic.CompareAndSwapUint32(&jobIsRunning, 0, 1) {
+		go func() {
+			publish(lbs)
+			atomic.StoreUint32(&jobIsRunning, 0)
+		}()
+	} else {
+		logger.Info("Request blocked!")
+		if retryTimer != nil {
+			retryTimer.Stop()
+		}
+		//Retry last blocked request after 0.5s
+		retryTimer = time.AfterFunc(time.Second/2, func() {
+			logger.Info("Retrying!")
+			publish(lbs)
+		})
+	}
+}
+
+func publish(lbs *Lbs) {
+
+	//Stop any retry request since new request appeared
+	if retryTimer != nil {
+		retryTimer.Stop()
+		retryTimer = nil
+	}
+
+	//
+	// Create battle peer list, for the third embed
+	//
+	battle := make(map[string]*battlePeers)
+
+	battlePeerCount := 0
+	var battlePeersIDs string
```

Comment:
use `map[string]bool`
This logic is incorrect and　string concatenation and retrieval are computationally intensive.

Reply:
[Done](https://github.com/inada-s/gdxsv/pull/66/files#diff-bbc193ec510092c18d1667da1c5232f4dfe8137c20b8f99db6b3c1fe606449cdR195)